packages/core/test/app/block-production/merkle-interval-tree.spec.ts

```diff
@@ -48,97 +69,97 @@ describe('merkle-index-tree', () => {
           )
       })
       it('should throw if left & right nodes are out of order', async () => {
-        const left = new MerkleIntervalTreeNode(
+        const left = new GenericMerkleIntervalTreeNode(
           Buffer.from([13]),
           Buffer.from([15])
         )
-        const right = new MerkleIntervalTreeNode(
+        const right = new GenericMerkleIntervalTreeNode(
           Buffer.from([31]),
           Buffer.from([10])
         )
-        const parentCall = () => MerkleIntervalTree.parent(left, right)
+        const parentCall = () => GenericMerkleIntervalTree.parent(left, right)
         parentCall.should.throw()
       })
     })
     it('should generate a generic tree', async () => {
       const leaves = []
       for (let i = 0; i < 4; i++) {
         leaves.push(
-          new MerkleIntervalTreeNode(
+          new GenericMerkleIntervalTreeNode(
             Buffer.from([Math.floor(Math.random() * 100)]),
             Buffer.from([i])
           )
         )
       }
-      const IntervalTree = new MerkleIntervalTree(leaves)
+      const IntervalTree = new GenericMerkleIntervalTree(leaves)
       log(IntervalTree.levels)
       log(IntervalTree.root)
     })
     it('should generate and verify inclusion proofs for generic tree', async () => {
       const leaves = []
       for (let i = 0; i < 4; i++) {
         leaves.push(
-          new MerkleIntervalTreeNode(
+          new GenericMerkleIntervalTreeNode(
             Buffer.from([Math.floor(Math.random() * 100)]),
             Buffer.from([i])
           )
         )
       }
-      const IntervalTree = new MerkleIntervalTree(leaves)
+      const IntervalTree = new GenericMerkleIntervalTree(leaves)
       const leafPosition = 3
       const inclusionProof = IntervalTree.getInclusionProof(leafPosition)
-      MerkleIntervalTree.verify(
+      GenericMerkleIntervalTree.verify(
         leaves[leafPosition],
-        leafPosition,
         inclusionProof,
         IntervalTree.root().hash
       )
     })
   })
   describe('MerkleStateIntervalTree', () => {
     it('should generate a tree without throwing', async () => {
-      const stateUpdates = []
-      for (let i = 0; i < 4; i++) {
-        const stateObject = new AbiStateObject(
-          '0xbdAd2846585129Fc98538ce21cfcED21dDDE0a63',
-          '0x123456'
-        )
-        const range = new AbiRange(
-          new BigNum(i * 100),
-          new BigNum((i + 0.5) * 100)
-        )
-        const stateUpdate = new AbiStateUpdate(
-          stateObject,
-          range,
-          new BigNum(1),
-          '0xbdAd2846585129Fc98538ce21cfcED21dDDE0a63'
-        )
-        stateUpdates.push(stateUpdate)
-      }
+      const stateUpdates = generateNSequentialStateUpdates(4)
       const merkleStateIntervalTree = new MerkleStateIntervalTree(stateUpdates)
       log('root', merkleStateIntervalTree.root())
     })
+    it('verification should throw if state update range intersects branch bounds', async () => {
```

Comment:
Nitpick but I'd change the test name from  "it(verification should throw if..." -> to
```
describe('verifyExectedRoot') {
  it('should throw if state update range...')
}
```

A little more grammatically correct